### ISSUE TITLE
Testing Font Awesome

### ISSUE THREAD

**Issue by ronm123** [thexerteproject/xerteonlinetoolkits#924]
@torinfo @JohnSmith-LT I was going to do a test of accessibility and Font Awesome with latest develop but also had been testing use of v4 and v5 FA icons anyway. I don't know when or how this happened but for me at least it seems that at the moment the v4 icons aren't showing when you play a XOT project - not even the navigation icons in the footer.

To replicate using latest develop:
Create a new XOT project or edit an existing project and ensure that the theme set is one that uses FA icons for the navigation buttons in the footer e.g. Future Teacher or RGBO.
Play the project - do you see the expected FA icons or just rectangular boxes? I just get empty boxes but this must be very recent?

Ideally the goal is that previous and v5 icons show when you play projects and ideally in the editor too and I know John has done some work on that but I'm not sure we have this fully working or indeed how we could have because at the moment the Bootstrap and Decision players only include v 4.3.0,

In my copy of develop looking at the players we have the following:
xot
`
    <link href="%TEMPLATEPATH%common_html5/font-awesome-5.6.3/css/font-awesome.min.css%VERSION_PARAM%" rel="stylesheet">
	<link href="%TEMPLATEPATH%common_html5/font-awesome-5.6.3/css/v4-shims.min.css%VERSION_PARAM%" rel="stylesheet">`

decision
`<link rel="stylesheet" href="%TEMPLATEPATH%common/font-awesome-4.3.0/css/font-awesome.min.css">`
site
 `<link rel="stylesheet" href="%TEMPLATEPATH%common/font-awesome-4.3.0/css/font-awesome.min.css">`

If I re-enable the link that is commented out then the footer buttons appear again but I'm not sure when that change was made? I know the inclusion of v4-shims.min.css after the 5.6.3 font-awesome.min.css is supposed to ensure that the earlier icons still work but that doesn't seem to be the case.
If I manually add v5 icons in an XOT project e.g. <i class="fab fa-500px"></i> they don't appear in the editor or when I play the project. 
Some of that is predictable because the current links to the 5.6.3 css are incorrect. They point to font-awesome whereas the actual folder name for 5.6.3 is currently fontawesome all one word without the hyphen. If I change those two lines then the manually added v5 icons and then v4 icons added via the editor show when playing the project but the footer icons still don't unless I un-comment the original line too.

Ideally we need to get v4 and v5 icons working in the editor and the player for the xot template and then repeat that for the Bootstrap and Decision templates too. The following is the source of a plain text page I've been using for testing:
Sorry that won't paste here correctly but I used examples from https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4
Do you get the same results as me?

**Comment by ronm123** [comment]
I had to edit my original message because it didn't like the pasted code. Also I forgot to mention that I'm unsure whether we should change the link paths or the actual folder names for the 5.6.3 css. I'm thinking it's better to just correct the links so that they correctly point to the folders we already have rather than add the hyphen back in to the folder name?
Also from what I read the first/main link to 5.6.3 should be to all.min.css not to font-awesome.min.css?
Also we currently include all the 5.6.3 folders e.g.
![image](https://user-images.githubusercontent.com/5167146/89640807-414ea580-d8a8-11ea-9558-b72711e7aaab.png)
Do we need to include them all or just a subset? The entire set is about 14MB+ and so much larger than the original two folders.
Finally we don't currently have the 5.6.3 folder in site or decision and I guess the easiest thing is to put the 5.6.3 folders in both those places but should we look at a 'common' folder of stuff for all templates and then ensure they are also included during export. I suspect yes we should but much easier just to replicate the folders for each template and not really a big issue doing that.

**Comment by torinfo** [comment]
Hi Ron,

Apparently I committed that yesterday. That should NOT have been committed yet, because I wanted to test this and make sure that the v5 icons would work in xot. They do not at the moment. Anyhow, I'll have a look and see what the best way to fix it is.

**Comment by torinfo** [comment]
@ronm123 I cleaned up the fontawesome 5 folder, and included the all.min.css. The folder is now 2.4 Mb. 

If I leave out the commented line, the icons in the footer do not work. That is because the icons are activated in css, with content. That should still work, provided the corret font is chosen, and the correct font-weight is chosen:

`#x_nextBtn:after { content: "\f138"; font-family: "Font Awesome 5 Free";  font-weight: 900;}`

So, there are two choices. Include the original fonteawesom and not change the themes, or adapt the themes and only include fontawesome 5.

**Comment by torinfo** [comment]
@ronm123 Adapting the themes easier than stated above. Just adapt

`#x_footerBlock button {
  font-family: "Font Awesome 5 Free", sans-serif;
  font-size: 1.9em;
  width: 1.5em;
  padding-right: 0;
  margin-left: 5px;
  font-weight:900;
}`

(Change the font and add font-weight)

**Comment by ronm123** [comment]
@torinfo trouble with that is what about all the themes that people may have adapted/created themselves and also the thmes they created via xhibit? I suggest we just re-enable keep the original folder if that works?

**Comment by torinfo** [comment]
@ronm123 We can do that as well of course. I'll commit that.

**Comment by torinfo** [comment]
Committed 1abb368b323e702dc3e2d568c19d9638686dce88

**Comment by ronm123** [comment]
This is fixed now so will close it. There is an accessibility issue with FA icons on some specific page types but I'll add that separately.
